keyvault/client/keys/src/main/java/com/azure/keyvault/keys/KeyClient.java

```diff
@@ -0,0 +1,639 @@
+     * </pre>
+     *
+     * @param keyImportOptions The key import configuration object containing information about the json web key being imported.
+     * @throws NullPointerException if {@code keyImportOptions} is {@code null}.
+     * @throws HttpRequestException if {@code name} is empty string.
+     * @return A {@link Response} whose {@link Response#value() value} contains the {@link Key imported key}.
+     */
+    public Response<Key> importKey(KeyImportOptions keyImportOptions) {
+        Objects.requireNonNull(keyImportOptions, "The key import configuration parameter cannot be null.");
+        KeyImportRequestParameters parameters = new KeyImportRequestParameters()
+                .key(keyImportOptions.keyMaterial())
+                .hsm(keyImportOptions.hsm())
+                .keyAttributes(new KeyRequestAttributes(keyImportOptions));
+        return service.importKey(endpoint, keyImportOptions.name(), API_VERSION, ACCEPT_LANGUAGE, parameters, CONTENT_TYPE_HEADER_VALUE).block();
+    }
+
+    /**
+     * Gets the public part of the specified key and key version. The get key operation is applicable to all key types and
+     * If the requested key is symmetric, then no key material is released in the response. This operation requires the {@code keys/get} permission.
+     *
+     * <p><strong>Code Samples</strong></p>
+     * <p>Gets a specific version of the key in the key vault. Prints out the details of the returned key.</p>
+     * <pre>
+     * String keyVersion = "6A385B124DEF4096AF1361A85B16C204";
+     * Key keyWithVersion = keyClient.getKey("keyName", keyVersion).value();
+     * System.out.printf("Key is returned with name %s and id %s \n", keyWithVersion.name(), keyWithVersion.id());
+     * </pre>
+     *
+     * @param name The name of the key, cannot be null
+     * @param version The version of the key to retrieve. If this is an empty String or null, this call is equivalent to calling {@link KeyClient#getKey(String)}, with the latest version being retrieved.
+     * @throws ResourceNotFoundException when a key with {@code name} and {@code version} doesn't exist in the key vault.
+     * @throws HttpRequestException if {@code name} or {@code version} is empty string.
+     * @return A {@link Response} whose {@link Response#value() value} contains the requested {@link Key key}.
+     */
+    public Response<Key> getKey(String name, String version) {
+        String keyVersion = "";
+        if (version != null) {
+            keyVersion = version;
+        }
+        return service.getKey(endpoint, name, keyVersion, API_VERSION, ACCEPT_LANGUAGE, CONTENT_TYPE_HEADER_VALUE).block();
+    }
+
+    /**
+     * Get the public part of the latest version of the specified key from the key vault. The get key operation is applicable to all key types and
+     * If the requested key is symmetric, then no key material is released in the response. This operation requires the {@code keys/get} permission.
```

Comment:
>If the requested key is symmetric, then no key material is released in the response. [](start = 7, length = 84)

Currently key vault doesn't support symmetric keys through the key api.

Reply:
excluded this from Javadoc.